### PR TITLE
Fixed duplicate filters by using value and label as key.

### PR DESCRIPTION
**Pull request by revanth0212**
## Description

There was an error in the filters list when there are multiple filters with the same label. Fixed it by using filter value along with the filter label as key.

## Related Issue
Closes [PWA-945](https://jira.corp.magento.com/browse/PWA-945)

### Verification Stakeholders
@dpatil-magento 

### Verification Steps

Backend URL - https://venia-smoketest-x3mnlqi-c5v7sxvquxwl4.eu-4.magentosite.cloud/

1. Go to the deployed app.
2. Search a string "Tops"
3. Now click on any result or try sort/filter. The page should render without any error.

## Checklist

- [x] I have added tests to cover my changes, if necessary.
- [ ] I have updated the documentation accordingly, if necessary.

### CODE REVIEW

File: packages/peregrine/lib/Price/__tests__/__snapshots__/price.spec.js.snap [hunk 1]
```diff
@@ -48,7 +48,7 @@ Array [
     1
   </span>,
   <span>
-     
+
```
Comment (revanth0212) [inline packages/peregrine/lib/Price/__tests__/__snapshots__/price.spec.js.snap:51]: Not sure why this changed. I just re-ran the tests and the snapshot was updated.

Comment (supernova-at) [inline packages/peregrine/lib/Price/__tests__/__snapshots__/price.spec.js.snap:51]: Not sure why this snapshot would have changed 🤔

File: packages/venia-ui/lib/components/FilterModal/FilterList/filterDefault.js [hunk 2]
```diff
@@ -7,13 +7,13 @@ import defaultClasses from './filterDefault.css';
...
     const { classes: propsClasses, isSelected, item, ...restProps } = props;
-    const { label } = item || {};
+    const { label, value_index } = item || {};
     const classes = mergeClasses(defaultClasses, propsClasses);
 
     return (
         <Checkbox
             classes={classes.root}
-            field={label}
+            field={`${label}-${value_index}`}
```
Comment (supernova-at) [inline packages/venia-ui/lib/components/FilterModal/FilterList/filterDefault.js:16]: We always have a `value_index`, right?

**Review by supernova-at — APPROVED** [review]

File: packages/peregrine/lib/Price/__tests__/__snapshots__/price.spec.js.snap (hunk 1 above)
Comment (supernova-at) [reply packages/peregrine/lib/Price/__tests__/__snapshots__/price.spec.js.snap:51]: Oh, I see you've already commented on this lol.

**Review by supernova-at — CHANGES_REQUESTED** [review]
Verification steps failed :(

Filtering worked but sorting and clicking on a search result brought up the same `keepState` error.

File: packages/venia-ui/lib/components/FilterModal/FilterList/filterDefault.js (hunk 2 above)
Comment (sirugh) [reply packages/venia-ui/lib/components/FilterModal/FilterList/filterDefault.js:16]: Would hope! We don't do any validation of this info. Might want to enforce prop types if they aren't.

File: packages/peregrine/lib/Price/__tests__/__snapshots__/price.spec.js.snap (hunk 1 above)
Comment (sirugh) [reply packages/peregrine/lib/Price/__tests__/__snapshots__/price.spec.js.snap:51]: Yea and tests failed now so...maybe its a node version thing? 🤷

**Comment by supernova-at** [comment]
Verification is working for me now ✅

**Review by supernova-at — APPROVED** [review]

**Review by supernova-at — COMMENTED** [review]
Verification fails for test backend.

**Review by supernova-at — DISMISSED** [review]
Verification fails for test backend.

Repro:
1. Set `MAGENTO_BACKEND_URL` to test backend
2. Search "tops"
3. Click "Leah Yoga Top"

Screen Shot 2020-09-28 at 2 12 28 PM

**Review by sirugh — DISMISSED** [review]
This works for me - I did not hit the same issue Andy did.

**Review by sirugh — APPROVED** [review]

**Comment by revanth0212** [comment]
> This works for me - I did not hit the same issue Andy did.

Me neither it works fine.

**Comment by dpatil-magento** [comment]
QA Approved.

**Pull request merged**